airflow/cli/commands/dag_command.py

```diff
@@ -375,15 +409,18 @@ def dag_list_dags(args, session=NEW_SESSION) -> None:
             file=sys.stderr,
         )
 
-    def get_dag_detail(dag: DAG) -> dict:
+    def get_dag_detail(dag: DAG) -> dict | None:
         dag_model = DagModel.get_dagmodel(dag.dag_id, session=session)
-        dag_detail = dag_schema.dump(dag_model)
+        if dag_model:
+            dag_detail = dag_schema.dump(dag_model)
+        else:
+            dag_detail = _get_dagbag_dag_details(dag)
         return {col: dag_detail[col] for col in valid_cols}
 
     AirflowConsole().print_as(
         data=sorted(dagbag.dags.values(), key=operator.attrgetter("dag_id")),
         output=args.output,
-        mapper=get_dag_detail,
+        mapper=get_dag_detail,  # type: ignore[arg-type]
```

Comment:
```suggestion
        mapper=get_dag_detail,
```
I don't think we need this anymore